File: LeetCode/Python/Largest Combination With Bitwise AND Greater Than Zero.py
"""
  Largest Combination With Bitwise AND Greater Than Zero
  URL: https://leetcode.com/problems/largest-combination-with-bitwise-and-greater-than-zero
  Difficulty: Medium
  Topic : Array, Hash Table, Bit Manipulation, Counting
"""

class Solution:
    def largestCombination(self, candidates: List[int]) -> int:
        maxLength = 0
        binaryString = []
        for c in candidates:
            br = "{0:b}".format(c)[::-1]
            maxLength = max(len(br), maxLength)
            binaryString.append(br)
        result = 0
        for i in range(maxLength):
            count = 0
            for b in binaryString:
                if len(b) > i and b[i] == "1":
                    count += 1
            result = max(result, count)
        return result

File: LeetCode/Medium/Largest Combination With Bitwise AND Greater Than Zero.swift
/*
  Largest Combination With Bitwise AND Greater Than Zero
  URL: https://leetcode.com/problems/largest-combination-with-bitwise-and-greater-than-zero
  Difficulty: Medium
  Topic : Array, Hash Table, Bit Manipulation, Counting
*/

class Solution {
    func largestCombination(_ candidates: [Int]) -> Int {
        var maxLength = 0
        let binaryString = candidates.map {
            let binary = String($0, radix:2)
            maxLength = max(maxLength, binary.count)
            return Array(binary.reversed())
        }
        return (0..<maxLength).reduce(into: 0) { r, i in
            r = max(r, binaryString.reduce(0) { $0 + ($1.count > i && $1[i] == "1" ? 1 : 0) })
        }
    }
}
